packages/block-library/src/post-content/edit.js

```diff
@@ -2,17 +2,38 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import PostContentInnerBlocks from './inner-blocks';
 
-export default function PostContentEdit( { context: { postId, postType } } ) {
-	if ( postId && postType ) {
+export default function PostContentEdit( {
+	context: { postId: contextPostId, postType: contextPostType },
+} ) {
+	const { id: currentPostId, type: currentPostType } = useSelect(
+		( select ) => select( 'core/editor' ).getCurrentPost() ?? {}
```

Comment:
Ideally, block-library shouldn't rely on `core/editor` store as a block can be used in several contexts. Is there a way to infer this from the block context API?

Reply:
Id argue this doesn't rely on `core/editor`, it just checks for it.  We are only concerned with applying this fix when the context Post and the current Post are the same, as that is what was creating the loop and crashing the editor.  So in the case that `core/editor` is not available, this continues to work as intended as this fix is not necessary in that circumstance.

Reply:
I can see that but do you think we can leverage the context API instead? It seems like a natural fit for this kind of checks?

Reply:
Oh, sure!  Im drawing a blank at the moment, do you have an example of where else this is used?

Reply:
Is that `useContext` from `@wordpress/element` ?

Reply:
I think this block context API: https://github.com/WordPress/gutenberg/tree/master/packages/block-editor/src/components/block-context

That said, we can't use any existing data which exists in context for this, but mabye we can add some extra data?

So we consume the context `postType` and `postId` here. So we use that to determine which post to render. The problem is that in the normal post editor, these values are set to the root post. If there exists a post content block in the root post, it will try to render itself recursively until it crashes the editor.

This is basically getting around that in the case that the post specified via context is the same as the root one in the editor store.

On any page with the core/editor store:
- The context postId and postType equal the one in the post/editor store (so we show the placeholder)

On the site editor:
- The context postId and postType are set, but the post/editor store does not specify anything at all since it isn't a post page. So we know it's ok to use them.

Instead of this, what if we instead add a bit of context to the root block context specifying the editor type? Then, if it is `post-editor`, we can show a placeholder, but if it is `site-editor`, we can safely render actual post content.

Reply:
> Instead of this, what if we instead add a bit of context to the root block context specifying the editor type? Then, if it is post-editor, we can show a placeholder, but if it is site-editor, we can safely render actual post content.

This seems like a good approach, curious if @youknowriad has any other thoughts.

Reply:
Mmm right! I guess I'd avoid a new context value, I thought the existing one was already sufficient but apparently not, so I'm fine with the current "hack" for the moment.